File: internal/app/locales.go
package app

import "strings"

type Locale struct {
	Code string
	Name string
}

func defaultLocales() []Locale {
	return []Locale{
		{Code: "ar", Name: "Arabic"},
		{Code: "bg", Name: "Bulgarian"},
		{Code: "cs", Name: "Czech"},
		{Code: "da", Name: "Danish"},
		{Code: "de", Name: "German"},
		{Code: "el", Name: "Greek"},
		{Code: "en", Name: "English"},
		{Code: "en-GB", Name: "English (UK)"},
		{Code: "en-US", Name: "English (US)"},
		{Code: "es", Name: "Spanish"},
		{Code: "es-419", Name: "Spanish (Latin America)"},
		{Code: "et", Name: "Estonian"},
		{Code: "fi", Name: "Finnish"},
		{Code: "fr", Name: "French"},
		{Code: "he", Name: "Hebrew"},
		{Code: "hi", Name: "Hindi"},
		{Code: "hr", Name: "Croatian"},
		{Code: "hu", Name: "Hungarian"},
		{Code: "id", Name: "Indonesian"},
		{Code: "it", Name: "Italian"},
		{Code: "ja", Name: "Japanese"},
		{Code: "ko", Name: "Korean"},
		{Code: "lt", Name: "Lithuanian"},
		{Code: "lv", Name: "Latvian"},
		{Code: "ms", Name: "Malay"},
		{Code: "nb", Name: "Norwegian Bokmal"},
		{Code: "nl", Name: "Dutch"},
		{Code: "pl", Name: "Polish"},
		{Code: "pt", Name: "Portuguese"},
		{Code: "pt-BR", Name: "Portuguese (Brazil)"},
		{Code: "pt-PT", Name: "Portuguese (Portugal)"},
		{Code: "ro", Name: "Romanian"},
		{Code: "ru", Name: "Russian"},
		{Code: "sk", Name: "Slovak"},
		{Code: "sl", Name: "Slovenian"},
		{Code: "sv", Name: "Swedish"},
		{Code: "th", Name: "Thai"},
		{Code: "tr", Name: "Turkish"},
		{Code: "uk", Name: "Ukrainian"},
		{Code: "vi", Name: "Vietnamese"},
		{Code: "zh-Hans", Name: "Chinese (Simplified)"},
		{Code: "zh-Hant", Name: "Chinese (Traditional)"},
	}
}

func localeLabel(locale Locale) string {
	if strings.TrimSpace(locale.Name) == "" {
		return locale.Code
	}
	return locale.Name + " (" + locale.Code + ")"
}

func localeNameByCode(locales []Locale) map[string]string {
	out := map[string]string{}
	for _, locale := range locales {
		out[locale.Code] = locale.Name
	}
	return out
}

File: src/locales.ts
export interface Locale {
  code: string;
  name: string;
}

export function defaultLocales(): Locale[] {
  return [
    { code: "ar", name: "Arabic" },
    { code: "bg", name: "Bulgarian" },
    { code: "cs", name: "Czech" },
    { code: "da", name: "Danish" },
    { code: "de", name: "German" },
    { code: "el", name: "Greek" },
    { code: "en", name: "English" },
    { code: "en-GB", name: "English (UK)" },
    { code: "en-US", name: "English (US)" },
    { code: "es", name: "Spanish" },
    { code: "es-419", name: "Spanish (Latin America)" },
    { code: "et", name: "Estonian" },
    { code: "fi", name: "Finnish" },
    { code: "fr", name: "French" },
    { code: "he", name: "Hebrew" },
    { code: "hi", name: "Hindi" },
    { code: "hr", name: "Croatian" },
    { code: "hu", name: "Hungarian" },
    { code: "id", name: "Indonesian" },
    { code: "it", name: "Italian" },
    { code: "ja", name: "Japanese" },
    { code: "ko", name: "Korean" },
    { code: "lt", name: "Lithuanian" },
    { code: "lv", name: "Latvian" },
    { code: "ms", name: "Malay" },
    { code: "nb", name: "Norwegian Bokmal" },
    { code: "nl", name: "Dutch" },
    { code: "pl", name: "Polish" },
    { code: "pt", name: "Portuguese" },
    { code: "pt-BR", name: "Portuguese (Brazil)" },
    { code: "pt-PT", name: "Portuguese (Portugal)" },
    { code: "ro", name: "Romanian" },
    { code: "ru", name: "Russian" },
    { code: "sk", name: "Slovak" },
    { code: "sl", name: "Slovenian" },
    { code: "sv", name: "Swedish" },
    { code: "th", name: "Thai" },
    { code: "tr", name: "Turkish" },
    { code: "uk", name: "Ukrainian" },
    { code: "vi", name: "Vietnamese" },
    { code: "zh-Hans", name: "Chinese (Simplified)" },
    { code: "zh-Hant", name: "Chinese (Traditional)" },
  ];
}

export function localeLabel(locale: Locale): string {
  if (!locale.name?.trim()) return locale.code;
  return `${locale.name} (${locale.code})`;
}

export function localeNameByCode(
  locales: Locale[],
): Record<string, string> {
  const out: Record<string, string> = {};
  for (const locale of locales) {
    out[locale.code] = locale.name;
  }
  return out;
}
